### ISSUE TITLE
`no-array-sort` should not insist cloning (using .toSorted()) a cloned array

### ISSUE THREAD

**Issue by mkahvi** [sindresorhus/eslint-plugin-unicorn#2735]
`no-array-sort` insists on using `.toSorted()` when chaining .

```js
const sortedSubset = someArray
  .filter(f => f.true)
  .sort((a,b) => b.sort - a.sort);
```
```js
const sorted = someArray
  .map(f => ({ ...f, enabled: f.value > 0 }))
  .sort((a,b) => b.sort - a.sort);
```

**Comment by fisker** [comment]
IMO, `.toSorted` should be preferred in this case.
If there are performance issues, engines should do some optimization.

**Comment by mkahvi** [comment]
Engines can't really optimize away explicit request for array copying.

**Comment by fregante** [comment]
Conceptually `return x.sort()` is terrible. Either you sort in place _or_ use its return value.

While I agree that `.toSorted()`'s cloning is wasteful in these cases, I do expect engines to optimize this instance. Alternatively, you might just want to `eslint-disable` the rule mentioning it's a performance optimization.

As always for performance optimizations: have you tested it? If it _is_ substantially slower, do report it to V8's and JSC's bug trackers
